hail/src/main/scala/is/hail/shuffler/LoggerCode.scala

```diff
@@ -0,0 +1,25 @@
+package is.hail.shuffler
+
+import is.hail.asm4s._
+import org.apache.log4j.Logger
+
+import scala.reflect.{ClassTag, classTag}
+
+object LoggerCode {
+  def getLogger[T: ClassTag](): Code[Logger] =
+    Code.invokeStatic1[Logger, String, Logger]("getLogger", classTag[T].runtimeClass.getName())
+}
+
+
+class LoggerCode(
```

Comment:
Small stylistic thing: we normally put Code first: CodeArray, CodeInt, RichCodeRegion, etc.

Reply:
Actually, I went with `...Value` since they all require a value.

It seems somewhat bad that we've bifurcated into `P...Code` & `P...Value` and `Code...` as the two naming schemes. I don't much care, but it seems bad to perpetuate, by adding `Value...`? Or maybe better to keep prefix style for non-Hail-objects?